src/Common/MemoryTracker.cpp

```diff
@@ -130,9 +169,10 @@ void MemoryTracker::allocImpl(Int64 size, bool throw_if_memory_exceeded, MemoryT
     {
         /// Since the MemoryTrackerBlockerInThread should respect the level, we should go to the next parent.
         if (auto * loaded_next = parent.load(std::memory_order_relaxed))
-            loaded_next->allocImpl(size, throw_if_memory_exceeded,
-                level == VariableContext::Process ? this : query_tracker);
-        return;
+            std::ignore = loaded_next->allocImpl(size, throw_if_memory_exceeded,
```

Comment:
I tried to avoid extra multiplication cause the formula looks like
`1.0 - (1.0 - sample_probability) * (1.0 - parent_sample_probability)`

But if it's ok I can do it